### PR TITLE
Adds React Native example app

### PR DESCRIPTION
**Pull request by wyattjoe40**
* Adds React Native example app
* The app runs on both iOS and Android
* There are two modules that can be used from inside React code: `AttentiveCreativeModule` (used to render a creative) and `AttentiveEventTrackingModule` (used to track user events).

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { RootStackParamList } from './navTypes';
+import HomeScreen from './HomeScreen'
+import ProductScreen from './ProductScreen'
+
+const Stack = createNativeStackNavigator<RootStackParamList>();
+
+const { AttentiveCreativeModule } = NativeModules;
+
+function App(): JSX.Element {
+  useEffect(() => {
```
Comment (oliviakim321) [inline src/App.tsx:21]: does this mean that we call identify every time the home screen is rendered? should we add a comment saying ideally we call identify every asap & then when we have new identifying data?

File: ios/AttentiveSDKSingleton.m [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@implementation AttentiveSDKSingleton
+
+static ATTNSDK* _sdk;
+
++ (id)sharedSDK {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        _sdk = [[ATTNSDK alloc] initWithDomain:@"YOUR_ATTENTIVE_DOMAIN" mode:@"production"];
+    });
+    return _sdk;
```
Comment (oliviakim321) [inline ios/AttentiveSDKSingleton.m:21]: 💯

**Review by oliviakim321 — APPROVED** [review]
LGTM!

File: src/App.tsx (hunk 1 above)
Comment (wyattjoe40) [reply src/App.tsx:21]: Nope, the second parameter to `useEffect` is an array of dependencies. Setting that dependency list to `[]` means the effect only runs once.

**Pull request merged**